src/python_testing/TC_AVSUM_2_7.py

```diff
@@ -0,0 +1,132 @@
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+# See https://github.com/project-chip/connectedhomeip/blob/master/docs/testing/python.md#defining-the-ci-test-arguments
+# for details about the block below.
+#
+# === BEGIN CI TEST ARGUMENTS ===
+# test-runner-runs:
+#   run1:
+#     app: ${CAMERA_APP}
+#     app-args: --discriminator 1234 --KVS kvs1 --trace-to json:${TRACE_APP}.json
+#     script-args: >
+#       --storage-path admin_storage.json
+#       --commissioning-method on-network
+#       --discriminator 1234
+#       --passcode 20202021
+#       --PICS src/app/tests/suites/certification/ci-pics-values
+#       --trace-to json:${TRACE_TEST_JSON}.json
+#       --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+#       --endpoint 1
+#     factory-reset: true
+#     quiet: true
+# === END CI TEST ARGUMENTS ===
+
+import chip.clusters as Clusters
+from chip.interaction_model import Status
+from chip.testing.matter_testing import MatterBaseTest, TestStep, default_matter_test_main, has_feature, run_if_endpoint_matches
+from TC_AVSUMTestBase import AVSUMTestBase
+
+
+class TC_AVSUM_2_7(MatterBaseTest, AVSUMTestBase):
+
+    def desc_TC_AVSUM_2_7(self) -> str:
+        return "[TC-AVSUM-2.7] DPTZSetViewport command validation"
+
+    def steps_TC_AVSUM_2_7(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "Commissioning, already done", is_commissioning=True),
+            TestStep(2, "Create a viewport smaller than the supported minimum"),
+            TestStep(3, "Send DPTZSetVieport with an unknown stream ID, verify NotFound response"),
+            TestStep(4, "Send a VideoStreamAllocate command to AVStreamManagement to allocate a video stream ID. Record the returned ID"),
+            TestStep(5, "Send DPTZSetVieport with a the viewport created in Step 2. Verify ConstraintError response"),
+            TestStep(6, "Create a viewport with valid AR that is larger than the camera sensor"),
+            TestStep(7, "Send DPTZSetVieport with a the viewport created in Step 6. Verify ConstraintError response"),
+            TestStep(8, "Create a valid viewport give the dimensions of the sensor and the device resolution"),
+            TestStep(9, "Send DPTZSetVieport with a the viewport created in Step 9. Verify success"),
+            TestStep(10, "Modify the valid viewport so that the AR is invalid"),
+            TestStep(11, "Send DPTZSetVieport with a the viewport created in Step 10. Verify ConstraintError response"),
+        ]
+        return steps
+
+    def pics_TC_AVSUM_2_7(self) -> list[str]:
+        pics = [
+            "AVSUM.S", "AVSM.S"
+        ]
+        return pics
+
+    @run_if_endpoint_matches(has_feature(Clusters.CameraAvSettingsUserLevelManagement,
```

Comment:
also requires the AV stream management cluster.

Reply:
Done